### DISCUSSION TITLE
AOC Day 9 Discussion - Fit to be tied

### DISCUSSION THREAD

**Discussion by rhdeck** [Virtual-Coffee/virtualcoffee.io#768] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 9](https://adventofcode.com/2022/day/9) here.

Don't forget about the VC Advent of Code private leaderboard:

Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
Use join code 363496-e6f42283

**Comment by rhdeck** [comment]
I got roped in by this one. Used numpy tho - that was fun! [solve](https://github.com/rhdeck/aoc-2022/blob/main/src/9/day_9.ipynb)

**Comment by danieltott** [comment]
Finally done! 
https://github.com/danieltott/aoc2022/blob/main/src/day09/index.ts

Was able to refactor after seeing part to using the same logic, just abstracted, which was nice.

Right up until my logic didn't turn out to be right 😂 

PR: 
https://github.com/danieltott/aoc2022/pull/6

**Reply by rhdeck** [reply]
[HULK NO LIKE RICKETY ROPE BRIDGE](https://playgroundai.com/post/clbh01axv1bzys601fmvm7ygx)

**Comment by doleraj** [comment]
Today just fucking destroyed me. I spent far too long thinking my code was working for the test cases, when I had in fact made two errors that were cancelling each other out - forgot to add the origin to the tail position list, and also was not only adding unique values, because - like an idiot - I forgot that Set doesn't work on arrays because `[1,2] !=== [1,2]`. 

Anyhow it's over and I hate everything. https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day09/index.ts

**Reply by tkshill** [reply]
This is me. I am this.

**Comment by tkshill** [comment]
Had a busy weekend so far. Here's 9. I really _like_ my solution but took forever to nail part 2 because I just got lost in the sauce I guess.

Got to use the forgotten sibling of reduce, scan, so that's neat.

[solve](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day09.fs)

**Comment by davidalpert** [comment]
again, imma have to study @tkshill's approach to this as mine took way more code, although on reflection 30 lines or so was printing it out so that my tests were visual and for the most part I like the way the rest reads

- in F# https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day09_rope_bridge.fs#L11-L181

**Reply by tkshill** [reply]
I found this person's attempts and whatever brevity you feel I have, he has [more](https://github.com/s-hirvonen/aoc/blob/master/2022/Day09.fsx)

But similarly to you, I still like my solutions. It's just fun seeing folks more concise.

And I like watching yours as examples of solutions with proper methodology. Your code captures your thought processes very well.
